### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by IMB11** [FabricMC/fabric-docs#239]
This can probably be done during the 1.21.4 updating process.

- [x] Version-related components aren't updating immediately in some cases, need to force a re-render when switching versions - <https://vitepress.dev/guide/ssr-compat#clientonly> potential fix?
- [x] ~~Investigate whether or not [Algolia Search](https://vitepress.dev/reference/default-theme-search#algolia-search) would be better than the current local search. Local search is becoming slow due to many locales and pages. It wasn't really designed for larger documentation. Even the official vitepress documentation (which is very large in itself) uses algolia. Can apply for an open source license (no limits) here: https://docsearch.algolia.com/apply~~ - Update: It seems #259 has improved performance for some reason, worth checking to see if the improvements have stayed steady down the road.

- [x] Try https://vitepress.dev/guide/mpa-mode, I dont have high hopes that it will work though.

**Comment by its-miroma** [comment]
> Version-related components aren't updating immediately in some cases

switching between two Versioned versions (eg. `1.21 -> 1.20.4`) does not refresh the page at all

 > Try [vitepress.dev/guide/mpa-mode](https://vitepress.dev/guide/mpa-mode)

I tried on my low-end system, and it didn't help me at all.
However, I personally favor MPA mode for UX, even with full page reloads (the same is happening in the 404 page right now)

**Comment by modmuss50** [comment]
I agree, MPA sounds like a good idea. I dont think there is a downside, the simpler the final output is the better.

**Comment by IMB11** [comment]
Been doing some testing, MPA is significantly faster in-dev, I'll be making a PR to address the rest of the issues, and also @its-miroma that was fixed by updating the versioning plugin to 1.3.0.
